src/modules/Elsa.Persistence.EntityFrameworkCore/Common/Store.cs

```diff
@@ -25,6 +25,13 @@ public async Task SaveAsync(TEntity entity, Func<TDbContext, TEntity, TEntity>?
         entity = onSaving?.Invoke(dbContext, entity) ?? entity;
         await dbContext.BulkInsertOrUpdateAsync(new[] { entity }, config => { config.EnableShadowProperties = true; }, cancellationToken: cancellationToken);
     }
+    
+    public async Task SaveAsync<TDto>(TEntity entity, TDto dto, Func<TDbContext, TEntity, TDto, TEntity>? onSaving = default, CancellationToken cancellationToken = default)
```

Comment:
The DTO thing seems like an odd duck in the nest. Can't we do the mapping from DTO to Entity in a higher layer, same approach as is done elsewhere?

Reply:
Unfortunately not without exposing DbContext to higher levels. I could also include the properties of dto in the entity and ignore them in EntityTypeConfiguration. But there was a problem with serializing in that way + I find it cleaner to separate dto and entity and keep entity as clean as possible(only with the properties we want to persist). 

I can expose the dbContext to higher level if you dont find this approach clean?